cmd/root_test.go

```diff
@@ -46,3 +46,50 @@ func TestValidConfig(t *testing.T) {
 
 	assert.Nil(t, cmd.Execute())
 }
+
+func TestTooManyCobraArgs(t *testing.T) {
+	tests := []struct {
+		name        string
+		args        []string
+		expectError bool
+	}{
+		{
+			name:        "Too many args",
+			args:        []string{"abc", "pqr", "xyz"},
+			expectError: true,
+		},
+		{
+			name:        "Too few args",
+			args:        []string{},
+			expectError: true,
+		},
+		{
+			name:        "One arg is okay",
+			args:        []string{"abc"},
+			expectError: false,
+		},
+		{
+			name:        "Two args is okay",
+			args:        []string{"abc", "pqr"},
+			expectError: false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			cmd, err := NewRootCmd(func(config cfg.Config) error { return nil })
+			if err != nil {
+				t.Fatalf("Error while creating the root command: %v", err)
+			}
+			cmd.SetArgs(tc.args)
+
+			err = cmd.Execute()
+
+			if tc.expectError {
```

Comment:
Can we pass expected errors in test cases:
`expectedErr: fmt.Errorf("accepts between %d and %d arg(s), received %d", 1, 2, 3),`

And do a normal assert:
assert.Equal(t, tc.expectedErr, err)

Reply:
I think we can rely on cobra to return a sensible error and not do that validation ourselves. If a future version of cobra changes the error message, the test shouldn't break. I can still add if you insist.